test/api/TradeAPI.spec.ts

```diff
@@ -175,4 +214,314 @@ describe('TradeAPI', () => {
       });
     });
   });
+
+  describe('#fetchTradeQuoteAsync', () => {
+    let subjectFromToken: Address;
+    let subjectToToken: Address;
+    let subjectFromTokenDecimals: number;
+    let subjectToTokenDecimals: number;
+    let subjectRawAmount: string;
+    let subjectFromAddress: Address;
+    let subjectSetToken: SetTokenAPI;
+    let subjectGasPrice: number;
+
+    beforeEach(async () => {
+      subjectFromToken = '0xAAAA15AA9B462ed4fC84B5dFc43Fd2a10a54B569';
+      subjectToToken = '0xBBBB262A92581EC09C2d522b48bCcd9E3C8ACf9C';
+      subjectFromTokenDecimals = 8;
+      subjectToTokenDecimals = 6;
+      subjectRawAmount = '5';
+      subjectFromAddress = '0xCCCC262A92581EC09C2d522b48bCcd9E3C8ACf9C';
+      subjectSetToken = <unknown>{ val: 'settoken' } as SetTokenAPI;
+      subjectGasPrice = 20;
+    });
+
+    async function subject(): Promise<TradeQuote> {
+      return await tradeAPI.fetchTradeQuoteAsync(
+        subjectFromToken,
+        subjectToToken,
+        subjectFromTokenDecimals,
+        subjectToTokenDecimals,
+        subjectRawAmount,
+        subjectFromAddress,
+        subjectSetToken,
+        subjectGasPrice
+      );
+    }
+
+    it('should call the TradeQuoter with correct params', async () => {
+      const expectedQuoteOptions = {
```

Comment:
nit: I usually like the expected values to be defined right before the expectation (line 272)